File: examples/tube.cpp
#include <string>
#include <sstream>
#include <unistd.h>
#include <iostream>

#include "hilma/types/mesh.h"
#include "hilma/types/polyline.h"
#include "hilma/ops/path.h"
#include "hilma/io/ply.h"

int main(int argc, char **argv) {

    hilma::Polyline polyline;
    int total = 1000;
    for (int i = 0; i < total; i++) {
        float a = glm::radians(i * 0.5f);
        polyline.addVertex(glm::vec3(cos(a*1.2), sin(a*2.), -cos(a*3.2) * sin(a*1.5)) * 10.0f);
    }

    float width[] = {1.f, 1.1f, 1.2f, 1.1f, 1.f, 1.1f, 1.2f, 1.3f, 1.2f, 1.1f,
                     1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f };
    hilma::Mesh mesh = hilma::toTube(polyline, width, 20, 18, true);

    mesh.computeNormals();
    hilma::savePly("tube.ply", mesh, false);

    return 1;
}

File: examples/tube.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

from hilma import Mesh, Polyline, tube, vec3, savePly
from math import radians, cos, sin

polyline = Polyline()
for i in range(1000):
    a = radians(i * 0.5)
    polyline.addVertex(vec3(cos(a*1.2), sin(a*2.), -cos(a*3.2) * sin(a*1.5)) * 10.0)

mesh = tube(polyline, 1.0, 18)
savePly("tube.ply", mesh, False)